### ISSUE TITLE
TelemetryConverter configuration throws InvalidOperationException using the provided appsettings sample

### ISSUE THREAD

**Issue by FZilli** [serilog-contrib/serilog-sinks-applicationinsights#196]
**Description**
Configuring the "telemetryConverter" appsettings section with the sample provided in the README ("telemetryConverter": "Serilog.Sinks.ApplicationInsights.TelemetryConverters.TraceTelemetryConverter, Serilog.Sinks.ApplicationInsights") causes a System.InvalidOperationException to be thrown.

![image](https://user-images.githubusercontent.com/50174433/172790014-18d0068c-1dd2-408f-8d15-7081e6996980.png)

**Reproduction**
_Program.cs_
```csharp
var configuration = new ConfigurationBuilder()
    .SetBasePath(Directory.GetCurrentDirectory())
    .AddJsonFile("appsettings.json")
    .Build();

var logger = new LoggerConfiguration()
    .ReadFrom.Configuration(configuration)
    .CreateLogger();

logger.Information("Hello, world!");

Console.ReadLine();
```

_appsettings.json_
```json
{
  "Serilog": {
    "Using": [
      "Serilog.Sinks.ApplicationInsights"
    ],
    "MinimumLevel": {
      "Default": "Debug",
      "Override": {
        "Microsoft": "Information"
      }
    },
    "WriteTo": [
      {
        "Name": "ApplicationInsights",
        "Args": {
          "restrictedToMinimumLevel": "Information",
          "telemetryConverter": "Serilog.Sinks.ApplicationInsights.TelemetryConverters.TraceTelemetryConverter, Serilog.Sinks.ApplicationInsights"
        }
      }
    ],
    "Enrich": [ "FromLogContext" ],
    "Properties": {
      "Application": "Sample"
    }
  }
}
```

**Relevant package, tooling and runtime versions**
 [net6.0]:
   Top-level Package                                        Requested   Resolved
   > Microsoft.Extensions.Configuration.FileExtensions      6.0.0       6.0.0
   > Microsoft.Extensions.Configuration.Json                6.0.0       6.0.0
   > Serilog                                                2.11.0      2.11.0
   > Serilog.Settings.Configuration                         3.3.0       3.3.0
   > Serilog.Sinks.ApplicationInsights                      3.1.0       3.1.0

   Transitive Package                                           Resolved
   > Microsoft.ApplicationInsights                              2.12.0
   > Microsoft.Extensions.Configuration                         6.0.0
   > Microsoft.Extensions.Configuration.Abstractions            6.0.0
   > Microsoft.Extensions.Configuration.Binder                  2.0.0
   > Microsoft.Extensions.DependencyInjection.Abstractions      2.0.0
   > Microsoft.Extensions.DependencyModel                       3.0.0
   > Microsoft.Extensions.FileProviders.Abstractions            6.0.0
   > Microsoft.Extensions.FileProviders.Physical                6.0.0
   > Microsoft.Extensions.FileSystemGlobbing                    6.0.0
   > Microsoft.Extensions.Options                               2.0.0
   > Microsoft.Extensions.Options.ConfigurationExtensions       2.0.0
   > Microsoft.Extensions.Primitives                            6.0.0
   > Microsoft.NETCore.Platforms                                1.1.0
   > Microsoft.NETCore.Targets                                  1.1.0
   > System.Diagnostics.DiagnosticSource                        4.6.0
   > System.IO                                                  4.3.0
   > System.Reflection                                          4.3.0
   > System.Reflection.Primitives                               4.3.0
   > System.Runtime                                             4.3.0
   > System.Runtime.CompilerServices.Unsafe                     6.0.0
   > System.Runtime.Handles                                     4.3.0
   > System.Runtime.InteropServices                             4.3.0
   > System.Text.Encoding                                       4.3.0
   > System.Text.Encodings.Web                                  6.0.0
   > System.Text.Json                                           6.0.0
   > System.Threading.Tasks                                     4.3.0

**Additional context**
Using this alternative configuration fixes the problem but it seems not right

"Serilog.**Sinks.ApplicationInsights**.Sinks.ApplicationInsights.TelemetryConverters.TraceTelemetryConverter, Serilog.Sinks.ApplicationInsights"

**Comment by david-x-chen** [comment]
I think you can take a look at this https://stackoverflow.com/questions/67672174/how-to-set-telemetryconfiguration-for-application-insights-for-serilog-via-appse. 
In it, you can find the configuration set as 
`"telemetryConverter": "Serilog.Sinks.ApplicationInsights.Sinks.ApplicationInsights.TelemetryConverters.TraceTelemetryConverter, Serilog.Sinks.ApplicationInsights"`

**Comment by FZilli** [comment]
> I think you can take a look at this https://stackoverflow.com/questions/67672174/how-to-set-telemetryconfiguration-for-application-insights-for-serilog-via-appse. In it, you can find the configuration set as `"telemetryConverter": "Serilog.Sinks.ApplicationInsights.Sinks.ApplicationInsights.TelemetryConverters.TraceTelemetryConverter, Serilog.Sinks.ApplicationInsights"`

Hello David, thanks for your reply!
I'm aware that adding an extra pair of "**Sinks.ApplicationInsights** makes the "telemetryConverter" work (see _Additional context_), but it doesn't seem right...
If this is the expected behaviour then the documentation should be updated.

**Comment by gagwithgaffer** [comment]
The suggested link to the stack overflow question above may work for a net core web app project, but using a net core worker service requires we use "Microsoft.ApplicationInsights.WorkerService" package instead of "Microsoft.ApplicationInsights.AspNetCore" package, and this doesnt have an option to set for "EnableActiveTelemetryConfigurationSetup = true", so what is the solution please? I cant update this sink to ver. 4+ until we have a solution that is explained clearly and people can follow, please help...

**Comment by spaasis** [comment]
Has this been fixed in 4.0? Our previous `Sinks.ApplicationInsights.Sinks.ApplicationInsights` broke after update and after removing the duplicate the app starts again normally

**Comment by nblumhardt** [comment]
Ah now I understand what's going on! Yes, thanks @spaasis - this is a breaking change in 4.0, the README is referring to the v4.0 configuration. Upgrading to v4 or later will fix this.

I've added the missing note to the "breaking changes" section of the release notes:

https://github.com/serilog-contrib/serilog-sinks-applicationinsights/releases/tag/v4.0.0
